mathics/builtin/atomic/symbols.py

```diff
@@ -59,36 +59,134 @@ def _get_usage_string(symbol, evaluation, is_long_form: bool):
     definition = evaluation.definitions.get_definition(symbol.name)
     ruleusage = definition.get_values_list("messages")
     usagetext = None
-    import re
 
     # First look at user definitions:
     for rulemsg in ruleusage:
         if rulemsg.pattern.expr.elements[1].__str__() == '"usage"':
             usagetext = rulemsg.replace.value
-    if usagetext is not None:
-        # Maybe, if htmltout is True, we should convert
-        # the value to a HTML form...
+
+    if not is_long_form and usagetext:
         return usagetext
-    # Otherwise, look at the pymathics, and builtin docstrings:
+
     builtins = evaluation.definitions.builtin
     pymathics = evaluation.definitions.pymathics
-    bio = pymathics.get(definition.name)
-    if bio is None:
-        bio = builtins.get(definition.name)
+    bio = pymathics.get(definition.name) or builtins.get(definition.name)
 
     if bio is not None:
-        if not is_long_form and hasattr(bio.builtin.__class__, "summary_text"):
-            return bio.builtin.__class__.summary_text
-        from mathics.doc.common_doc import XMLDOC
+        from mathics.doc.doc_entries import DocumentationEntry
 
         docstr = bio.builtin.__class__.__doc__
         title = bio.builtin.__class__.__name__
         if docstr is None:
-            return None
-        usagetext = XMLDOC(docstr, title).text(0)
+            return usagetext
```

Comment:
Code involving mathics documentation should not be here. Please move this into mathics.doc.

Reply:
OK, I put it in a new module inside `mathics.doc` maybe we can rename it later.

Reply:
Most likely this will get removed altogether when we get documentation under control.